internal/middleware/compression_test.go

```diff
@@ -31,6 +31,67 @@ import (
 	"go.opentelemetry.io/collector/testutil"
 )
 
+func TestHTTPClientCompression(t *testing.T) {
+	testBody := []byte("uncompressed_text")
+	compressedBody, _ := compressGzip(testBody)
+
+	tests := []struct {
+		name     string
+		encoding string
+		reqBody  []byte
+	}{
+		{
+			name:     "NoCompression",
+			encoding: "",
+			reqBody:  testBody,
+		},
+		{
+			name:     "ValidGzip",
+			encoding: "gzip",
+			reqBody:  compressedBody.Bytes(),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				body, err := ioutil.ReadAll(r.Body)
+				require.NoError(t, err, "failed to read request body: %v", err)
+				assert.EqualValues(t, tt.reqBody, body)
+				w.WriteHeader(200)
+			})
+
+			addr := testutil.GetAvailableLocalAddress(t)
+			ln, err := net.Listen("tcp", addr)
+			require.NoError(t, err, "failed to create listener: %v", err)
+			srv := &http.Server{
```

Comment:
Anything against https://golang.org/pkg/net/http/httptest/#Server ? Looks more suitable for this case here.

Reply:
Nothing against :-) I just reused the approach used by the existing test in this same file. Can be refactored in a separate PR.

Reply:
I assume we are good here, resolving. Please reopen if needed.